sdk/include/psdk/winbase.h

```diff
@@ -3156,8 +3174,79 @@ SetFirmwareEnvironmentVariableW(
   _In_reads_bytes_opt_(nSize) PVOID pValue,
   _In_ DWORD nSize);
 
+#ifdef UNICODE
+#define SetFirmwareEnvironmentVariable SetFirmwareEnvironmentVariableW
+#else
+#define SetFirmwareEnvironmentVariable SetFirmwareEnvironmentVariableA
 #endif
 
+#endif
+
+#if (_WIN32_WINNT >= 0x0602)
+
+_Success_(return > 0)
+WINBASEAPI
+DWORD
+WINAPI
+GetFirmwareEnvironmentVariableExW(
+    _In_ LPCWSTR lpName,
+    _In_ LPCWSTR lpGuid,
+    _Out_writes_bytes_to_opt_(nSize, return) PVOID pBuffer,
+    _In_ DWORD nSize,
+    _Out_opt_ PDWORD pdwAttribubutes);
+
+_Success_(return > 0)
+WINBASEAPI
+DWORD
+WINAPI
+GetFirmwareEnvironmentVariableExA(
+    _In_ LPCSTR lpName,
+    _In_ LPCSTR lpGuid,
+    _Out_writes_bytes_to_opt_(nSize, return) PVOID pBuffer,
+    _In_ DWORD nSize,
+    _Out_opt_ PDWORD pdwAttribubutes);
+
+#ifdef UNICODE
+#define GetFirmwareEnvironmentVariableEx GetFirmwareEnvironmentVariableExW
+#else
+#define GetFirmwareEnvironmentVariableEx GetFirmwareEnvironmentVariableExA
+#endif
+
+WINBASEAPI
+BOOL
+WINAPI
+SetFirmwareEnvironmentVariableExW(
+    _In_ LPCWSTR lpName,
+    _In_ LPCWSTR lpGuid,
+    _In_reads_bytes_opt_(nSize) PVOID pValue,
+    _In_ DWORD nSize,
+    _In_ DWORD dwAttributes);
+
+WINBASEAPI
+BOOL
+WINAPI
+SetFirmwareEnvironmentVariableExA(
+    _In_ LPCSTR lpName,
+    _In_ LPCSTR lpGuid,
+    _In_reads_bytes_opt_(nSize) PVOID pValue,
+    _In_ DWORD nSize,
+    _In_ DWORD dwAttributes);
+
+#ifdef UNICODE
+#define SetFirmwareEnvironmentVariableEx SetFirmwareEnvironmentVariableExW
+#else
+#define SetFirmwareEnvironmentVariableEx SetFirmwareEnvironmentVariableExA
+#endif
+
+_Success_(return != FALSE)
```

Comment:
Already understood because it's BOOL as return type, and it's not a function where FALSE means "success"...

Reply:
Superfluous SAL notations are removed.

Reply:
> Already understood because it's BOOL as return type, and it's not a function where FALSE means "success"...

Something I found:
`BOOL` is `int`, and has no builtin `_Success_` annotation like `NTSTATUS` and `HRESULT`.
```
typedef int                 BOOL;
```
MS also use `_Success_` to evaluate `BOOL` or `int` return value, see [Best practices and examples (SAL) - TRUE in _Success_](https://learn.microsoft.com/en-us/cpp/code-quality/best-practices-and-examples-sal?view=msvc-170#true-in-_success_):
>If the function succeeds when the return value is nonzero, use return != 0 as the success condition instead of return == TRUE

And some API in Windows SDK have `_Success_(return != FALSE)` annotation, like:
```
WINBASEAPI
_Success_(return != FALSE)
BOOL
WINAPI
GetSystemTimeAdjustment(
    _Out_ PDWORD lpTimeAdjustment,
    _Out_ PDWORD lpTimeIncrement,
    _Out_ PBOOL lpTimeAdjustmentDisabled
    );
```
But, seems MSVC compiler didn't concerned about `_Success_` annotation for a `int` return value, no warning on code analysis or compile-time if I ignore it:
```
DWORD dw1, dw2, dw3;
GetSystemTimeAdjustment(&dw1, &dw2, &dw3);
return dw1 + dw2 + dw3;
```
or
```
char sz[5];
sprintf_s(sz, _countof(sz), "xxxxxxxxxxxxxxxxxxxxxxx");
printf(sz);
```

However, `_Success_(return != FALSE)` is unnecessary for now.